src/window/action_popup.c

```diff
@@ -156,28 +147,32 @@ static void handle_input(const mouse *m, const hotkeys *h)
         window_go_back();
     }
 
-    if (generic_buttons_handle_mouse(mouse_in_dialog(m), 80, 40, buttons, data.show_cancel_button ? 4 : 3, &data.focus_button_id)) {
+    if (generic_buttons_handle_mouse(mouse_in_dialog(m), 80, 40, buttons, 5, &data.focus_button_id)) {
         return;
     }
-
 }
 
 static void button_select_option(int option, int param2)
 {
-    if (option == 1 || option == 2 || option == 3) {
-        data.close_func(option);
-        window_go_back();
-    } else if (!option) {
-        data.close_func(option);
-        window_go_back();
+    if (option && option != CONFIRM_BUTTON) {
+        data.selected_policy = option;
+    } else {
+        if (option == CONFIRM_BUTTON) {
+            if (data.selected_policy) {
```

Comment:
You should confirm if the new selected policy is different than the previous selected policy, otherwise the player will pay for effectively no change in policy.

For extra points, make "Apply policy" grayed out and unselectable if the selected policy is the already active policy.

Reply:
Yes I need to know current policy to prevent cost deduction when selecting current one.